apollo-router/tests/integration/subscriptions/ws_passthrough.rs

```diff
@@ -544,3 +560,208 @@ async fn test_subscription_ws_passthrough_pure_error_payload_with_coprocessor()
+
+    assert_eq!(total_active, 1);
+    assert_eq!(total_active + total_terminating, 1);
+
+    match verify_subscription_events(stream, expected_events, true)
+        .await
+        .map_err(|e| format!("Event verification failed: {}", e))
+    {
+        Ok(_events) => {}
+        Err(err) if err.contains("unexpected EOF") => {
+            log::warn!("{}", err)
+        }
+        Err(err) => return Err(err.into()),
+    }
+
+    router.graceful_shutdown().await;
+    // router.assert_shutdown().await;
+
+    // Check for errors in router logs
+    router.assert_log_not_contained("connection shutdown exceeded, forcing close");
+
+    info!(
+        "✅ Passthrough subscription mode test completed successfully with {} events",
+        custom_payloads.len()
+    );
+
+    Ok(())
+}
+
+#[tokio::test(flavor = "multi_thread")]
+async fn test_subscription_ws_passthrough_on_schema_reload() -> Result<(), BoxError> {
+    if !graph_os_enabled() {
+        eprintln!("test skipped");
+        return Ok(());
+    }
+
+    // Create fixed payloads for consistent testing
+    let custom_payloads = vec![create_user_data_payload(1), create_user_data_payload(2)];
+    let interval_ms = 10;
+
+    // Start subscription server with fixed payloads, but do not terminate the connection
+    let (ws_addr, http_server) =
+        start_subscription_server_with_payloads(custom_payloads.clone(), interval_ms, false).await;
+
+    // Create router with port reservations
+    let mut router = IntegrationTest::builder()
+        .supergraph("tests/integration/subscriptions/fixtures/supergraph.graphql")
+        .config(include_str!(
+            "fixtures/subscription_schema_reload.router.yaml"
+        ))
+        .build()
+        .await;
+
+    // Configure URLs using the string replacement method
+    let ws_url = format!("ws://{}/ws", ws_addr);
+    router.replace_config_string("http://localhost:{{PRODUCTS_PORT}}", &http_server.uri());
+    router.replace_config_string("http://localhost:{{ACCOUNTS_PORT}}", &ws_url);
+    router.replace_config_string("rng:", "accounts:");
+
+    info!("WebSocket server started at: {}", ws_url);
+
+    router.start().await;
+    router.assert_started().await;
+
+    // Use the configured query that matches our server configuration
+    let query = create_sub_query(interval_ms, custom_payloads.len());
+    let (_, response) = router.run_subscription(&query).await;
+
+    // Expect the router to handle the subscription successfully
+    assert!(
+        response.status().is_success(),
+        "Subscription request failed with status: {}",
+        response.status()
+    );
+
+    let stream = response.bytes_stream();
+    let expected_events = vec![
+        create_initial_empty_response(),
+        create_expected_user_payload(1),
+        create_expected_user_payload(2),
+        create_expected_reload_payload(),
+    ];
+
+    // try to reload the config file
+    router.replace_schema_string("createdAt", "created");
```

Comment:
is this still a valid schema?

Reply:
should we also check that this works as expected if it's the part of the supergraph we do interact with, or that doesn't matter?

Reply:
It doesn't matter because we close the subscription and we make sure in our tests it's not using the old pipeline